modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterManager.java

```diff
@@ -1040,6 +1050,32 @@ public RasterManager(
                 }
             }
         }
+        if (heterogeneousCRS) {
+            // If the reader is kept open (and the rasterManager doesn't change)
+            // it would be useful to cache the result of a query for
+            // a specific EPSG code on the index, so that a DB access won't be
+            // repeated while the info is in cache.
+
+            alternativeCRSCache =
+                    CacheBuilder.newBuilder()
+                            .maximumSize(50)
+                            .expireAfterWrite(60, TimeUnit.SECONDS)
```

Comment:
Magic number, best to make it a system variable if not a read parameter.

Reply:
I was originally planning to use a system variable indeed. 
Forgot to do at the end. Done, thx